eventlet/wsgi.py

```diff
@@ -479,6 +479,9 @@ def handle_one_response(self):
         use_chunked = [False]
         length = [0]
         status_code = [200]
+        # Status code of 1xx or 204 or 2xx to CONNECT request MUST NOT send body and related headers
+        # https://httpwg.org/specs/rfc7230.html#rfc.section.3.3.1
+        bodyless = [False]
```

Comment:
It bugs me a little that we're not actually _enforcing_ the absence of a body -- if application code responds `204 No Content` but then also returns an app iter that has some bytes, we're going to yield them out and mess up message framing. Bit of a slippery slope to go down, though, I suppose.

Reply:
Yes, agreed, thank you. But also we have a funny hack to allow random header case for compatibility with some popular server.

IMHO, we should claim WSGI and HTTP are separated at eventlet.wsgi, it takes care of transfer headers, may drop body and proper application expects that anyway. But also allow things like 204 with body via feature flag.

But it seems a fair amount of change. Suggest to extract this as separate task and implement later.

Reply:
Agreed; it can certainly be a follow-up -- or we just tell application developers to be more careful. Definitely not a blocker to getting this fix merged.